src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.*;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final String LINE_SEPARATOR = "\n";
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)  {
+        StringBuilder builder = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo).append(LINE_SEPARATOR);
+
+        LocalDate from = LocalDate.now();
+        LocalDate to = LocalDate.now();
+
+        try {
+            from = LocalDate.parse(dateFrom, formatter);
+            to = LocalDate.parse(dateTo, formatter);
+
+        } catch (DateTimeParseException e) {
+            System.out.println("Error parsing date " + e);
+        }
+
+        List<String> employeeNameByKey = new ArrayList<>(Arrays.stream(names).toList());
+        List<Integer> employeeSumByKey = new ArrayList<Integer>(Collections.nCopies(names.length, 0));
+
+        for (int i = 0; i < data.length; i++) {
+            String[] splittedString = data[i].split(" ");
+            LocalDate paymentDate = LocalDate.now();
+
+            try {
+                paymentDate = LocalDate.parse(splittedString[0], formatter);
+            } catch (DateTimeParseException e) {
+                System.out.println("Error parsing date in names" + e);
```

Comment:
The error message 'Error parsing date in names' is misleading. The issue is with parsing the payment date, not the names. Consider revising the message for clarity.